app/middleware/soaprest/middleware.go

```diff
@@ -63,8 +63,10 @@ func (s *soapREST) Middleware(next http.Handler) http.Handler {
 		// Reference: https://pkg.go.dev/net/http#Request
 		newReq.ContentLength = -1
 		newReq.Body = io.NopCloser(bytes.NewReader(jsonBody))
+		newReq.Header.Set("Accept", "application/json")
 		newReq.Header.Set("Content-Type", "application/json")
 		newReq.Header.Set("X-SOAP-Action", action)
+		newReq.Header.Set("X-Content-Type", r.Header.Get("Content-Type"))
```

Comment:
Variable `mt` is better than Get("Content-Type")

charset is not relevant to upstreams and action params are available via X-SOAP-Action.

```suggestion
		newReq.Header.Set("X-Content-Type", mt)
```

Reply:
I agree with your comment. 
Modified the code to use `mt` only for X-Content-Type header instead of full Content-Type.